cli/src/commands/split.rs

```diff
@@ -208,3 +201,34 @@ the operation will be aborted.
     tx.finish(ui, format!("split commit {}", commit.id().hex()))?;
     Ok(())
 }
+
+// Rebases the children of `original_commit` by replacing `original_commit` with
+// `new_siblings`. Any parents other than `original_commit` will remain after
+// the rebase.
+fn rebase_children_for_siblings_split(
+    tx: &mut WorkspaceCommandTransaction,
+    settings: &UserSettings,
+    original_commit: &Commit,
+    new_siblings: Vec<Commit>,
+) -> Result<usize, CommandError> {
+    let children: Vec<Commit> = RevsetExpression::commit(original_commit.id().clone())
+        .children()
+        .evaluate_programmatic(tx.base_repo().as_ref())?
+        .iter()
+        .commits(tx.base_repo().store())
+        .try_collect()?;
+    let mut num_rebased = 0;
+    for child in children {
+        let mut new_parents = new_siblings.clone();
+        new_parents.extend(
+            child
+                .parents()
+                .into_iter()
+                // Filter out the commit being split.
+                .filter(|c| c.id() != original_commit.id()),
+        );
```

Comment:
In order to preserve the parent order better, I think this should be using `flat_map()` where we replace the old parent by the new ones and leave unrelated parents unchanged.

Reply:
I thought about this issue when I wrote it but I didn't know about `flat_map`. This is great. Thanks!